packages/tablev2/Util.tsx

```diff
@@ -0,0 +1,161 @@
+  onRelativeXChange: (i: number) => void;
+}) => {
+  const [x, setX] = React.useState(0);
+
+  const updateMousePosition = (e: MouseEvent) => {
+    setX(e.clientX - initialX);
+  };
+
+  const onMouseUp = (e: MouseEvent) => {
+    props.onRelativeXChange(e.clientX - initialX);
+    setX(0);
+    document.removeEventListener("mousemove", updateMousePosition);
+    document.removeEventListener("mouseup", onMouseUp);
+  };
+  const onMouseDown = (e: E) => {
+    initialX = e.clientX;
+    document.addEventListener("mousemove", updateMousePosition);
+    document.addEventListener("mouseup", onMouseUp);
+  };
+
+  return (
+    <div
+      onMouseDown={onMouseDown}
+      className={className}
+      style={{ transform: `translate(${x}px, 0)` }}
+      role="button"
+    >
+      <div>
+        <Icon
+          shape={SystemIcons.ResizeHorizontal}
+          size={RESIZE_ICON_SIZE}
+          color={dt.textColorSecondary}
+        />
+      </div>
+    </div>
+  );
+};
+
+///////////////////////////////////////////////////////////////////////////////
+//                            HEADER CELL CONTENT                            //
+///////////////////////////////////////////////////////////////////////////////
+export const HeaderCellContent = ({
+  columnHeader,
+  columnId,
+  headerTooltipContent
+}: {
+  columnId: string;
+  headerTooltipContent?: React.ReactNode;
+  columnHeader?: React.ReactNode;
+}) =>
+  headerTooltipContent ? (
+    <Flex gutterSize="xxs" className={style.cellFlexWrapper}>
+      <FlexItem>
+        <div className={textTruncate}>{columnHeader}</div>
+      </FlexItem>
+      <FlexItem flex="shrink">
+        <Tooltip
+          id={`${columnId}-tooltip`}
+          trigger={
+            <Icon
+              shape={SystemIcons.CircleInformation}
+              size={dt.iconSizeXs}
+              color={dt.greyLightDarken2}
+            />
+          }
+        >
+          {headerTooltipContent}
+        </Tooltip>
+      </FlexItem>
+    </Flex>
+  ) : (
+    <div className={textTruncate}>{columnHeader}</div>
+  );
+
+///////////////////////////////////////////////////////////////////////////////
+//                                  SORTERS                                  //
+///////////////////////////////////////////////////////////////////////////////
+
+export type SorterFactory<T> = <B extends string, A extends Record<B, T>>(
+  prop: B
+) => Sorter<A>;
+
+const string: SorterFactory<string | null | undefined> = prop => (_, dir) => (
+  a,
+  b
+) => {
+  // null/undefined sort after anything else
+  if (a[prop] == null) {
+    return 1;
+  }
+  if (b[prop] == null) {
+    return -1;
+  }
+  return dir * (a[prop] || "")!.localeCompare((b[prop] || "")!);
+};
+
+const number: SorterFactory<number | null | undefined> = prop => (_, dir) => (
+  a,
+  b
+) => {
+  // null/undefined sort after anything else
+  if (a[prop] == null) {
+    return 1;
+  }
+  if (b[prop] == null) {
+    return -1;
+  }
```

Comment:
@brandonc thanks a lot!

if we don't compare strictly here, wouldn't `0`s be somewhere amongst the null/undefineds?

Reply:
no, i was fooled by the falsiness concept once again 🤦‍♂